option optimize_for = LITE_RUNTIME;

// Network identification information
message mnetwork {
  // Network id
  required string id = 1;
  // Human-readable Network Name
  required string name = 2;
  enum etype {
    // first item id
    FIRST = 0;
    // Wireless Local Area Network
    WLAN = 0;
    // 3G Network
    UMTS = 1;
    // last item (guard value)
    LAST = 2;
  }
  // network status
  enum estatus {
    // detected while scanning
    DETECTED = 0;
    // detected and is current connection
    CONNECTED = 1;
    // vanished from scan, but kept in hopes of reappearing
    VANISHED = 2;
  }
  // Network type
  required etype type = 3;
  // Measurement series, one for each measured parameter.
  repeated mseries all_series = 4;
  // Network status
  required estatus status = 5;
  // Time of existence (number of calls to signalstren during
  // which the network was still present in the scan)
  required int32 time_of_existence = 6;
  // Vanishing timer: number of calls to signalstren during
  // while network status is still `VANISHED'.
  // Reset when status is changed from VANISHED to DETECTED
  // or CONNECTED.
  required int32 vanishing_timer = 7;
}

// Represents a series of values of some parameter.
message mseries {
  // Name of the measured parameter
  required string parameter = 1;
  
  // Count of processed measures
  required int32 size = 2;
  // Last processed measure
  required double last_value = 3;
  
  // Next 4 variables are for iterative calculation of mean and variance:
  required double old_m = 4;
  required double old_s = 5;
  required double new_m = 6;
  required double new_s = 7;
}

// The whole persisted state is here
message mnetinfo {
  // list of all detected networks
  repeated mnetwork all_networks = 1;

  // date & time of last call to signalstren in standard ISO format
  required string last_call_time = 2;
}
